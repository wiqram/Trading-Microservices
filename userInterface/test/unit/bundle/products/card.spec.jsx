import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import ProductCard from '../../../../src/bundle/product/card';

describe('ProductCard Snapshot Test', () => {
    const product = {
        productName: 'Saving A/C',
        currentBalance: 11.11,
        accountNumber: '111111',
        sortCode: '121212',
        availableBalance: 1212.12

    }
    it('renders correctly with all props given', () => {
        const wrapper = create(<BrowserRouter><ProductCard product={product} /> </BrowserRouter>)
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});