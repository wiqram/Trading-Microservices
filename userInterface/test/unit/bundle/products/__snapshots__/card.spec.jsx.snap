// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Snapshot Test renders correctly with all props given 1`] = `
Array [
  <div
    className="styles__Row-sc-1ldttm7-2 style__Card-sc-1rp0p1n-0 kkqgzJ"
  >
    <div
      className="styles__Col-sc-1ldttm7-1 bTiKQz"
    >
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__ProductName-sc-1rp0p1n-1 lgSegm"
        >
          Saving A/C
        </div>
        <div
          className="style__AccountNumber-sc-1rp0p1n-3 ghNEoX"
        >
          111111/121212
        </div>
      </div>
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__Currency-sc-1rp0p1n-5 XQCQa"
          size={36}
        >
          <div
            className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
          >
            £11.11
          </div>
        </div>
        <div
          className="style__BalanceLabel-sc-1rp0p1n-4 gFnvle"
        >
          Current Balance
        </div>
      </div>
      <div
        className="styles__Row-sc-1ldttm7-2 style__CardRow-sc-1rp0p1n-2 hSSDTv"
      >
        <div
          className="style__Currency-sc-1rp0p1n-5 bqdoCa"
          size={22}
        >
          <div
            className="currency__TextStyle-sc-1uazvwe-0 kMHNch"
          >
            £1,212.12
          </div>
        </div>
        <div
          className="style__BalanceLabel-sc-1rp0p1n-4 gFnvle"
        >
          Available Balance
        </div>
      </div>
    </div>
    <div
      className="styles__Col-sc-1ldttm7-1 style__NavItems-sc-1rp0p1n-6 bozNBC"
    >
      <a
        href="/product/undefined"
        onClick={[Function]}
      >
        <button
          className="styles__Button-sc-1ldttm7-3 styles__ButtonPrimary-sc-1ldttm7-4 jkYupd"
        >
          More Details
        </button>
      </a>
    </div>
  </div>,
  " ",
]
`;
